src/backend/dx11/src/device.rs

```diff
@@ -1764,12 +1797,15 @@ impl hal::Device<Backend> for Device {
         I::Item: Borrow<(&'a Memory, R)>,
         R: RangeArg<u64>,
     {
+        let _scope = debug::DebugScope::with_name(&self.context, &format!("FlushMappedRanges"));
 
         // go through every range we wrote to
         for range in ranges.into_iter() {
             let &(memory, ref range) = range.borrow();
             let range = memory.resolve(range);
 
+
+            let _scope = debug::DebugScope::with_name(&self.context, &format!("Range({:?})", range));
```

Comment:
Unfortunate thing is that the string is constructed even if debug scopes are disabled. Sounds like the creation of a scope needs to be a macro itself

Reply:
I'd hope that the compiler could optimize it out, but haven't actually checkedi if that's the case. Do you mean a macro that simply does `#[cfg(...)]` on the function call?

Reply:
I mean a macro like `debug_scope!("tralalal {:?}", something)`